lib/jazzy/config.rb

```diff
@@ -220,6 +220,11 @@ def expand_path(path)
         SourceDeclaration::AccessControlLevel.from_human_string(acl)
       end
 
+    config_attr :lint,
+      command_line: '--[no-]lint',
+      description: "Generate linter output.",
+      default: false
```

Comment:
Perhaps something like this would be sufficient:

    jazzy --skip-doc-generation --undocumented-symbols=/tmp/symbols.tsv

Benefits of this approach:

- We don't have to change error codes.
- Absence of `--undocumented-symbols` implies no generation of them at all.
- `--skip-doc-generation` skips the entire doc generation step, rather than simulating their execution like `--dry-run` implies.

Then we can drop the concept of `dry-run` and `lint` altogether. Thoughts?